snap/component.go

```diff
@@ -41,16 +54,74 @@ type ComponentSideInfo struct {
 	Revision  Revision            `json:"revision"`
 }
 
+// NewComponentSideInfo creates a new ComponentSideInfo.
+func NewComponentSideInfo(cref naming.ComponentRef, rev Revision) *ComponentSideInfo {
+	return &ComponentSideInfo{
+		Component: cref,
+		Revision:  rev,
+	}
+}
+
+// ComponentPlaceInfo holds information about where to put a component
+// in the system. It implements ContainerPlaceInfo.
+type ComponentPlaceInfo struct {
+	ComponentSideInfo
+	// SnapInstance and SnapRevision identify the snap that uses this component.
+	SnapInstance string
+	SnapRevision Revision
+}
+
+var _ ContainerPlaceInfo = (*ComponentPlaceInfo)(nil)
+
+// NewComponentPlaceInfo creates a new ComponentPlaceInfo.
+func NewComponentPlaceInfo(csi *ComponentSideInfo, instanceName string, snapRev Revision) *ComponentPlaceInfo {
+	return &ComponentPlaceInfo{
+		ComponentSideInfo: *csi,
+		SnapInstance:      instanceName,
+		SnapRevision:      snapRev,
+	}
+}
+
+// ContainerName returns the full name of the component.
+func (c *ComponentPlaceInfo) ContainerName() string {
+	return c.Component.String()
```

Comment:
this might not be right as it should contain the instance name of the snap, not the snap name?

Reply:
Changed